src/Breadcrumbs.php

```diff
@@ -147,77 +170,159 @@ public function links(array $value): self
     }
 
     /**
-     * The HTML attributes for the widgets nav container tag.
+     * Sets the HTML attributes for the list of items in the breadcrumbs.
      *
-     * {@see Html::renderTagAttributes()} for details on how attributes are being rendered.
+     * @param array $values Attribute values indexed by attribute names.
+     *
+     * @return self A new instance with the specified attributes for the list of items in the breadcrumbs.
+     *
+     * @see {\Yiisoft\Html\Html::renderTagAttributes()} for details on how attributes are being rendered.
      */
-    public function navOptions(array $value): self
+    public function listAttributes(array $values): self
     {
         $new = clone $this;
-        $new->navOptions = $value;
+        $new->listAttributes = $values;
 
         return $new;
     }
 
     /**
-     * The HTML attributes for the widget container tag. The following special options are recognized.
+     * Sets the ID of the list of items in the breadcrumbs.
+     *
+     * @param bool|string $value The ID of the breadcrumb component. If `true`, an ID will be generated automatically.
      *
-     * {@see Html::renderTagAttributes()} for details on how attributes are being rendered.
+     * @return self A new instance with the specified ID for the list of items in the breadcrumbs.
      */
-    public function options(array $value): self
+    public function listId(bool|string $value): self
     {
         $new = clone $this;
-        $new->options = $value;
+        $new->listId = $value;
 
         return $new;
     }
 
     /**
-     * The name of the breadcrumb container tag.
+     * Sets the HTML tag to be used for the list of items in the breadcrumbs.
      *
-     * @psalm-param non-empty-string $value
+     * @param string $value The HTML tag name for the list of items in the breadcrumbs.
+     *
+     * @return self A new instance class with the specified list tag name.
      */
-    public function tag(string $value): self
+    public function listTagName(string $value): self
     {
         $new = clone $this;
-        $new->tag = $value;
+        $new->listTagName = $value;
 
         return $new;
     }
 
+    public function render(): string
+    {
+        $attributes = $this->attributes;
+        $attributes['aria-label'] ??= 'breadcrumb';
+
+        if ($this->links === []) {
+            return '';
+        }
+
+        $list = $this->renderList();
+
+        if ($list === '') {
+            return '';
+        }
+
+        return Nav::tag()->addAttributes($attributes)->content("\n", $list, "\n")->encode(false)->render();
+    }
+
+    private function renderList(): string
+    {
+        $listAttributes = $this->listAttributes;
+        $classes = $listAttributes['class'] ?? null;
+
+        $listId = match ($this->listId) {
+            true => $listAttributes['id'] ?? Html::generateId(self::LIST_NAME . '-'),
+            '', false => null,
```

Comment:
```suggestion
            false => null,
```

Reply:
If, i remove the check for white space it generates, which is an error.

```html
<nav aria-label="breadcrumb">
<ol id class="breadcrumb">
 <li class="breadcrumb-item"><a href="/">Home</a></li>
 <li class="breadcrumb-item"><a href="#">Library</a></li>
 <li class="breadcrumb-item active" aria-current="page">Data</li>
 </ol>
 </nav>
```